nevergrad/optimization/recastlib.py

```diff
@@ -72,7 +73,42 @@ def _optimization_function(
         while remaining > 0:  # try to restart if budget is not elapsed
             options: tp.Dict[str, tp.Any] = {} if weakself.budget is None else {"maxiter": remaining}
             # options: tp.Dict[str, tp.Any] = {} if self.budget is None else {"maxiter": remaining}
-            if weakself.method == "CmaFmin2":
+            if weakself.method == "NLOPT":
+                # This is NLOPT, used as in the PCSE simulator notebook.
+                # ( https://github.com/ajwdewit/pcse_notebooks ).
+                import nlopt
+
+                def nlopt_objective_function(*args):
+                    data = np.asarray([arg for arg in args])[0]
+                    assert len(data) == weakself.dimension, (
+                        str(data) + " does not have length " + str(weakself.dimension)
+                    )
+                    if weakself._normalizer is not None:
+                        data = weakself._normalizer.backward(np.asarray(data, dtype=np.float32))
+                    return objective_function(data)
+
+                opt = nlopt.opt(nlopt.LN_SBPLX, weakself.dimension)
+                # Assign the objective function calculator
+                opt.set_min_objective(nlopt_objective_function)
+                # Set the bounds.
+                opt.set_lower_bounds([0.0] * weakself.dimension)
+                opt.set_upper_bounds([1.0] * weakself.dimension)
```

Comment:
does it have to be lists? arrays would be better
```suggestion
                opt.set_lower_bounds(np.zeros(weakself.dimension))
                opt.set_upper_bounds(np.ones(weakself.dimension))
```